crates/weaver_live_check/src/advice.rs

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Builtin advisors
+
+use std::{collections::BTreeMap, path::PathBuf};
+
+use serde::Serialize;
+use serde_json::Value;
+use weaver_checker::{
+    violation::{Advice, AdviceLevel, Violation},
+    Engine,
+};
+use weaver_forge::{jq, registry::ResolvedGroup};
+use weaver_resolved_schema::attribute::Attribute;
+use weaver_semconv::{
+    attribute::{AttributeType, PrimitiveOrArrayTypeSpec, TemplateTypeSpec, ValueSpec},
+    deprecated::Deprecated,
+    stability::Stability,
+};
+
+use crate::{live_checker::LiveChecker, Error, SampleRef};
+
+/// Embedded default live check rego policies
+pub const DEFAULT_LIVE_CHECK_REGO: &str =
+    include_str!("../../../defaults/policies/live_check_advice/otel.rego");
+
+/// Default live check rego policy path - used in error messages
+pub const DEFAULT_LIVE_CHECK_REGO_POLICY_PATH: &str =
+    "defaults/policies/live_check_advice/otel.rego";
+
+/// Embedded default live check jq preprocessor
+pub const DEFAULT_LIVE_CHECK_JQ: &str = include_str!("../../../defaults/jq/advice.jq");
+
+/// Provides advice on a sample
+pub trait Advisor {
+    /// Provide advice on a sample
+    fn advise(
+        &mut self,
+        sample: &SampleRef<'_>,
```

Comment:
This should be able to take `SampleRef<'_>` directly.  It's a common Rust pattern where you have a `{thing}Ref` struct that is efficient to just send as a value.